types/options.go

```diff
@@ -32,29 +32,38 @@ const (
 )
 
 var (
-	defaultStoreOptionsOnce sync.Once
+	defaultStoreOptionsOnce    sync.Once
+	loadDefaultStoreOptionsErr error
 )
 
-func loaddefaultStoreOptions() {
+func loadDefaultStoreOptions() {
 	defaultStoreOptions.RunRoot = defaultRunRoot
 	defaultStoreOptions.GraphRoot = defaultGraphRoot
 	defaultStoreOptions.GraphDriverName = ""
 
 	if path, ok := os.LookupEnv(storageConfEnv); ok {
 		defaultOverrideConfigFile = path
-	}
-
-	if _, err := os.Stat(defaultOverrideConfigFile); err == nil {
+		if err := ReloadConfigurationFileIfNeeded(path, &defaultStoreOptions); err != nil {
+			loadDefaultStoreOptionsErr = err
+			return
+		}
+	} else if _, err := os.Stat(defaultOverrideConfigFile); err == nil {
 		// The DefaultConfigFile(rootless) function returns the path
 		// of the used storage.conf file, by returning defaultConfigFile
 		// If override exists containers/storage uses it by default.
 		defaultConfigFile = defaultOverrideConfigFile
-		ReloadConfigurationFileIfNeeded(defaultOverrideConfigFile, &defaultStoreOptions)
+		if err := ReloadConfigurationFileIfNeeded(defaultOverrideConfigFile, &defaultStoreOptions); err != nil {
+			loadDefaultStoreOptionsErr = err
+			return
+		}
 	} else {
 		if !os.IsNotExist(err) {
 			logrus.Warningf("Attempting to use %s, %v", defaultConfigFile, err)
 		}
-		ReloadConfigurationFileIfNeeded(defaultConfigFile, &defaultStoreOptions)
+		if err := ReloadConfigurationFileIfNeeded(defaultConfigFile, &defaultStoreOptions); err != nil {
+			loadDefaultStoreOptionsErr = err
+			return
+		}
```

Comment:
This is a breaking change for us in CRI-O (see https://github.com/cri-o/cri-o/runs/7335875736?check_suite_focus=true). We do not require any storage.conf per default and would like to stick to the in-memory defaults. Was this behavioral change intentional?